src/dist/replication/lib/backup/replica_backup_manager.cpp

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) 2017-present, Xiaomi, Inc.  All rights reserved.
+// This source code is licensed under the Apache License Version 2.0, which
+// can be found in the LICENSE file in the root directory of this source tree.
+
+#include "replica_backup_manager.h"
+
+#include <dsn/dist/fmt_logging.h>
+#include <dsn/utility/filesystem.h>
+#include <dsn/dist/replication/replication_app_base.h>
+
+namespace dsn {
+namespace replication {
+
+// returns true if this checkpoint dir belongs to the policy
+static bool is_policy_checkpoint(const std::string &chkpt_dirname, const std::string &policy_name)
+{
+    std::vector<std::string> strs;
+    utils::split_args(chkpt_dirname.c_str(), strs, '.');
+    // backup_tmp.<policy_name>.* or backup.<policy_name>.*
+    return strs.size() >= 2 &&
+           (strs[0] == std::string("backup_tmp") || strs[0] == std::string("backup")) &&
+           strs[1] == policy_name;
+}
+
+// get all backup checkpoint dirs which belong to the policy
+static bool get_policy_checkpoint_dirs(const std::string &dir,
+                                       const std::string &policy,
+                                       /*out*/ std::vector<std::string> &chkpt_dirs)
+{
+    chkpt_dirs.clear();
+    // list sub dirs
+    std::vector<std::string> sub_dirs;
+    if (!utils::filesystem::get_subdirectories(dir, sub_dirs, false)) {
+        derror("list sub dirs of dir {} failed", dir.c_str());
+        return false;
+    }
+
+    for (std::string &d : sub_dirs) {
+        std::string dirname = utils::filesystem::get_file_name(d);
+        if (is_policy_checkpoint(dirname, policy)) {
+            chkpt_dirs.push_back(std::move(dirname));
+        }
+    }
+    return true;
+}
+
+void replica_backup_manager::on_cold_backup_clear(const backup_clear_request &request)
+{
```

Comment:
This function is only called by secondary replica, I suggest that add comment to show that, and add status check in this function. 
Besides, I notice that secondary replica don't clear checkpoints, only reset context. I remember that secondary replica may generate checkpoints in some cases, at lease I can find out this logic in former code.